icu4c/source/test/intltest/itrbnf.cpp

```diff
@@ -2489,6 +2490,21 @@ IntlTestRBNF::doLenientParseTest(RuleBasedNumberFormat* formatter, const char* t
     }
 }
 
+void
+IntlTestRBNF::TestNumberingSystem() {
```

Comment:
NOTE: This test is not duplicated on the Java side because there is no NUMBERING_FORMAT_STYLE (or any other RBNF-related style constants) in NumberFormat, so you can't pass those types to NumberFormat.getInstance() like you can in C.  (I checked, and the style parameter is an int, but the relevant constants on RuleBasedNumberFormat don't have distinct numeric values from the ones on NumberFormat, so they're clearly not intended to be used with NumberFormat.getInstance() ).